jar-connector/src/main/java/com/sixsq/slipstream/connector/ConnectorBase.java

```diff
@@ -294,18 +294,31 @@ protected String getPrivateSshKeyFileName() throws ConfigurationException, Valid
         return privateSshKeyFile;
     }
 
+    public static String getServerPublicSshKeyFilename() throws ConfigurationException, ValidationException {
+		return Configuration.getInstance().getProperty(ServiceConfiguration.CLOUD_CONNECTOR_ORCHESTRATOR_PUBLICSSHKEY);
+    }
+
     private String getUserPublicSshKey(User user) throws ValidationException, IOException {
 		return user.getParameter(
 				ExecutionControlUserParametersFactory.CATEGORY + "." + UserParametersFactoryBase.SSHKEY_PARAMETER_NAME)
 				.getValue();
     }
 
+    private String getUserOrServerPublicSshKey(User user) throws ValidationException, IOException {
```

Comment:
Yes, the reason is that some Clouds will check the format of the key and fail if the format is wrong.
2 others solutions:
- Generating a new keypair each time (need JSch as a dependency)
- Generating a keypair on my computer and then destroy the private key and hardcode the public key in SlipStream.

@loomis What do you think ?